### DISCUSSION TITLE
JS 코드와 HTML 마크업은 어떤 기준으로 분리하시나요?

### DISCUSSION THREAD

**Discussion by lokba** [woowacourse-study/2022-code-review-study#62] [Youtube] [answered]
질문이 이해하기 어려웠을텐데요.

1. index.html이 텅 비어있는 경우, JS가 모든 화면을 담당하는 경우
2. 사용자의 액션과 상관없이 항상 존재해야 하는 마크업은 html에 놔두고, 사용자의 액션에 따라 변하는 화면만 JS가 DOM에 접근하는 경우
3. 사용자의 액션에 따라 변하는 화면이지만, 상태가 변하지 않은 마크업은 html에 놔두는 경우. 

예를 들어 보겠습니다. 유튜브 미션에서는 `검색 결과 모달창`이 존재합니다.
`검색 결과 모달창`은 on/off 스위치에 따라 모달창이 show/hide하게 됩니다. 
하지만, 모달창 자체의 큰 골격(검색 입력창, 입력 버튼 등등)은 액션에 따라 변하지 않은 요소입니다.

`검색 결과 모달창`의 경우로 1번 2번 3번에 적용해보겠습니다.

1. index.html에는 `검색 결과 모달창` 마크업은 존재하지 않는다.
2. `검색 결과 모달창`은 사용자의 액션에 따르기에, index.html에 존재하지 않는다.
3. `검색 결과 모달창`의 변하지 않은 요소(검색 입력창, 버튼)은 index.html에 존재한다.

어떤 방식을 사용하고, 사용 이유와 다른 방식과 비교했을 때 얻는 장단점을 공유해주세요.

**Comment by byhhh2** [comment]
저는 1번의 방식을 사용할 것 같습니다..

유튜브 미션에서는 그럴 일 없겠지만, 특정 사용자가 `검색 결과 모달창`을 전혀 열어볼 일이 없다면 불필요하게 index.html에 존재하고 있는 것 아닐까요?🥲

**Comment by wonsss** [comment]
1번과 2번의 차이가 무엇인가요?

저는 2번의 `검색 결과 모달창은 사용자의 액션에 따르기에, index.html에 존재하지 않는다.`의 방식이 나은 것 같습니다.

이유는 무비와 비슷하구요.

그런데 실제 서비스들은 어떻게 하고 있는지 궁금해서 인스타그램 웹사이트에서 확인해봤는데, 2번처럼 동적으로 주입되는 것 같네요.

- 새 게시물 추가 버튼 클릭 전
![before](https://user-images.githubusercontent.com/59413128/160829361-a9c86a72-2b70-4b98-986c-da7556ae78b9.jpg)

- 새 게시물 추가 버튼 클릭 후 (모달창 표시됨)
	- 해당 마크업이 새로 추가됨 	
![after](https://user-images.githubusercontent.com/59413128/160829345-7f9a1f03-fbbe-4af5-a000-33c39b9de456.jpg)

**Comment by moonheekim0118** [comment, marked as answer]
나그네의 말 
> 이 록바의 의견은 초기 페이지거나
> 처음 접속했을때로 가정하고
> 고민하는게 더 좋아요
> app.js가 20mb라면
> 20mb가 다 읽힐때까지 화면이 렌더링 안됩니다


> 애초에
> html을
> js에서 그리는 것 자체가 말이 안되죠
> 역할을
> 초월한거임

**Reply by wonsss** [reply]
사업관점에서, 서비스 관점에서 SSR이 무조건 낫다.

**Reply by lokba** [reply]
페이지 소스 보기를 자주 확인해보자

**Reply by moonheekim0118** [reply]
미리 그릴지 아닐지 그걸 구분하는게 프론트엔드 개발자
